test/acceptance/tests/validation/validation_test.go

```diff
@@ -0,0 +1,906 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ecs"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/require"
+)
+
+// TestVolumeVariable tests passing a list of volumes to mesh-task.
+// This validates a big nested dynamic block in mesh-task.
+func TestVolumeVariable(t *testing.T) {
+	t.Parallel()
+	volumes := []map[string]interface{}{
+		{
+			"name": "my-vol1",
+		},
+		{
+			"name":      "my-vol2",
+			"host_path": "/tmp/fake/path",
+		},
+		{
+			"name":                        "no-optional-fields",
+			"docker_volume_configuration": map[string]interface{}{},
+			"efs_volume_configuration": map[string]interface{}{
+				"file_system_id": "fakeid123",
+			},
+		},
+		{
+			"name": "all-the-fields",
+			"docker_volume_configuration": map[string]interface{}{
+				"scope":         "shared",
+				"autoprovision": true,
+				"driver":        "local",
+				"driver_opts": map[string]interface{}{
+					"type":   "nfs",
+					"device": "host.example.com:/",
+					"o":      "addr=host.example.com,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2,noresvport",
+				},
+			},
+			"fsx_windows_file_server_volume_configuration": map[string]interface{}{
+				"file_system_id": "fakeid456",
+				"root_directory": `\\data`,
+				"authorization_config": map[string]interface{}{
+					"credentials_parameter": "arn:aws:secretsmanager:us-east-1:000000000000:secret:fake-fake-fake-fake",
+					"domain":                "domain-name",
+				},
+			},
+		},
+	}
+
+	terraformOptions := &terraform.Options{
+		TerraformDir: "./terraform/volume-variable",
+		Vars:         map[string]interface{}{"volumes": volumes},
+		NoColor:      true,
+	}
+	t.Cleanup(func() {
+		_, _ = terraform.DestroyE(t, terraformOptions)
+	})
+	terraform.InitAndPlan(t, terraformOptions)
```

Comment:
Should we be validating the output of the plan? This test only checks that the plan works (which is a good start) but it doesn't verify that the plan matches our expectations.

This is already a huge PR so maybe this can be addressed in a follow up?

Reply:
Most of the validation tests here were just moved to this new location (which is great 👌). So agreed, but I feel like improvements to the existing validation tests can wait for a follow up.